Data/Pool.hs

```diff
@@ -377,7 +410,8 @@ tryTakeResource pool@Pool{..} = do
       (Entry{..}:es) -> writeTVar entries es >> return (return . Just $ entry)
       [] -> do
         used <- readTVar inUse
-        if used == maxResources
+        currentMaximum <- readTVar maxResources
+        if used >= currentMaximum
```

Comment:
In which cases can `used` be greater than the `currentMaximum`? Ideally, shouldn't we have some check to never exceed the maximum?

Reply:
@codingkarthik so the max resources is a dynamic mutable variable now. Even after initializing the pool, one can decrease the max resources. In that case, the `used` would be greater than `currentMaximum`

Reply:
@codingkarthik so the max resources is a dynamic mutable variable now. Even after initializing the pool, one can decrease the max resources. In that case, the `used` would be greater than `currentMaximum`

Reply:
So, if after initializing the pool if there is a decrease in the max resources there can be more `used` connections than the `maxResources`? This will get normalized when the pool is trimmed, correct?

Reply:
@codingkarthik yes, that's correct.

Reply:
I've now tried to clarify this with a few comments and docstrings.